lib/parsers/moov_parser.rb

```diff
@@ -48,12 +48,14 @@ def call(io)
     width, height = parse_dimensions(decoder, atom_tree)
 
     # Try to find the "topmost" duration (respecting edits)
-    if mdhd = decoder.find_first_atom_by_path(atom_tree, 'moov', 'mvhd')
-      timescale = mdhd.field_value(:tscale)
-      duration = mdhd.field_value(:duration)
+    if mvhd = decoder.find_first_atom_by_path(atom_tree, 'moov', 'mvhd')
+      timescale = mvhd.field_value(:tscale)
+      duration = mvhd.field_value(:duration)
       media_duration_s = duration / timescale.to_f
     end
 
+    frame_rate = parse_sample_atom(decoder, atom_tree).truncate(2)
```

Comment:
Would this work for non-video files, at this level?

Reply:
it would still work, I tried it with the m4a test file in our fixtures but you still have a point and moved this a little bit further when we know for sure we are dealing with a video.